src/UnoFlipController.java

```diff
@@ -53,29 +51,50 @@ public void actionPerformed(ActionEvent e) {
                     null,
                     null);
 
+            // numPlayers used to create the amount of players the user has selected
             int numPlayers = 0;
-            if (result == JOptionPane.OK_OPTION) {
+            if (result == JOptionPane.OK_OPTION)
+            {
+                // Parse the input given by the user from the drop-down to use to create players
                 String option = (String) comboBox.getSelectedItem();
                 numPlayers = Integer.parseInt(option);
+                // Tell model the number of players in the game
                 this.model.setNumPlayers(numPlayers);
-            } else {
+            }
+            else // Exit the game if the user cancels
+            {
                 System.exit(0); // absolutely must select an option, otherwise, do not start the game
             }
 
-            for (int i = 0; i < numPlayers; i++){
+            // For the number of players prompt the user for name input
+            for (int i = 0; i < numPlayers; i++)
+            {
                 String name = (String) JOptionPane.showInputDialog("Enter player " + Integer.toString(i+1) + " name");
-                if (name == null || name.equals("")){
+
+                // If player does not select name, give a default name (Player + player number) ex. Player 2
+                if (name == null || name.equals(""))
+                {
                     this.model.createPlayer("Player " + (i+1));
-                } else{
+                }
+                else
+                {
                     this.model.createPlayer(name);
                 }
             }
+            // Start the game with model setUpInitialTopCard
             this.model.setUpInitialTopCard();
-        } else if (e.getActionCommand().equals(UnoFlipViewFrame.WILD_CMD)) {
+        }
+
+        // Wild command that prompts the user to select a colour
+        else if (e.getActionCommand().equals(UnoFlipViewFrame.WILD_CMD))
+        {
+            // Colour options for Player to chose from
             String[] numPlayerOptions = {"RED", "BLUE", "YELLOW", "GREEN"};
```

Comment:
This block of code to have a user select somethign using a JOptionPane.showOptionDialog() is repeated and will likely continue to repeat when we implement the dark side. I think it could be a good idea to make a private method that does this, for example: String openOptionDialog(String[] options). This can take in an array of strings as options and return the selected option. I understand that the behaviour is slightly different in this instance (where you use an infinite loop to repeatedly prompt) and the previous instance of this with the number of players (terminate system if cancel), but maybe we can change the number of players option dialog to exhibit the same behaviour as this one, where you continuously ask until they answer. This way, we can use a private method for both of these cases, and future ones when the dark side is implemented.